README.md

```diff
@@ -86,155 +87,144 @@ _choose one_ of the following submodule-cloning techniques:
+  git submodule update --init --recursive
   ```
 
-> **Note:** At any time during development
+> [!NOTE]
+> At any time during development
 > you can use the `git submodule` command to refresh submodules:
 >
 > ```terminal
-> $ git pull; git submodule update --init --recursive
+> git pull && git submodule update --init --recursive
 > ```
 
-## Setting up your local environment and serve changes
+## Set up your local environment and serve changes
 
 1. _Optional:_ After cloning the repo and its submodules,
    create a branch for your changes:
 
    ```terminal
-   $ git checkout -b <BRANCH_NAME>
+   git checkout -b <BRANCH_NAME>
    ```
 
-2. If the Docker Desktop application isn't already running on your machine,
-   start it. Look for the Docker status icon: if it has an exclamation
-   point (`!`), then update Docker Desktop before proceeding.
+2. From the root directory of the repository,
+   fetch the site's Dart dependencies.
+
+   ```terminal
+   dart pub get
+   ```
 
-3. Run the initial local setup command:
+3. From the root directory of the repository,
+   enable `corepack` to set up `pnpm`, then
+   fetch the site's npm dependencies.
+   If you already have `pnpm` installed,
+   you can skip the `corepack` commands.
 
    ```terminal
-   $ make setup
+   corepack enable
+   corepack install
+   pnpm install
    ```
 
-4. Serve the site locally (via `docker-compose`):
+4. From the root directory, serve the site locally.
 
    ```terminal
-   $ make up
+   dart run dart_site serve
    ```
 
-   The site is generated, and then the development server runs in the
-   Docker container, with the generated `_site` directory visible locally
-   as a mirrored volume from inside the container.
+   This command generates and serves the site on a
+   local port that's printed to your terminal.
 
-5. View your changes in the browser by navigating to `http://localhost:4000`.
-   > **Note:** Unless you're editing files under `site-shared`,
-   > you can safely ignore `ERROR: directory is already being watched` messages.
-   > For details, see [#1363](https://github.com/flutter/website/issues/1363).
+5. View your changes in the browser by navigating to <http://localhost:4000>.
 
-6. Make your changes to the local repo.
+   > [!NOTE]
+   > The port might be different if `4000` is taken.
+   
+   To instead view the source of generated site files,
+   check the `_site` directory.
 
-   The site will rebuild and the browser will autoreload to reflect the changes.
+6. Make your changes to the local repo.
 
-   > **Tip:** If you aren't seeing the changes you expect (e.g. src/_data),
-   > <kbd>Ctrl</kbd> + <kbd>C</kbd> out of your running dev server and rebuild the site from scratch
-   > using the following commands:
-   >
-   > ```terminal
-   > $ make down && make clean && make up
-   > ```
+   The site should automatically rebuild on most changes, but if
+   something doesn't update, exit the process and rerun the command.
```

Comment:
Are there "known issues" here that we can flag for folks?

Reply:
alternatively, an issue link or something

Reply:
I'd like to see which issues come up naturally, but I'll leave a comment to open an issue. Good point!